wagtail/admin/action_menu.py

```diff
@@ -270,8 +268,6 @@ def __init__(self, request, **kwargs):
         self.context = kwargs
         self.context["request"] = request
         page = self.context.get("page")
-        user_page_permissions = UserPagePermissionsProxy(self.request.user)
-        self.context["user_page_permissions"] = user_page_permissions
```

Comment:
This is necessary if we want to raise a warning for `UserPagePermissionsProxy`.

Hopefully this isn't a painful breaking change... I've added upgrade consideration notes.

Reply:
Perhaps the way forward is to tweak `UserPagePermissionsProxy` so that `permissions` is a cached_property, rather than an attribute that gets populated on `__init__` (potentially running an unwanted query), and we only raise the warning when a method of UserPagePermissionProxy is actually called.

(It would be nice if we could achieve that by putting the warning inside the `permissions` method body, but unfortunately that doesn't quite work due to us shortcutting that for superusers... so we'd probably have to stick it in every method individually instead.)

That way, we can leave the UserPagePermissionsProxy object in the context here for the duration of the deprecation period, adding basically no overheads, but still raising the warning if it actually gets used.

Reply:
I did consider this, but was hesitant due to the fact that we'd need to move the deprecation warning to each method.

That said, I think it's worth to make the deprecation less painful. The object has been reinstated in cb8cb935245d65c39a0204b10a56cebabf593cc6 and the deprecation warning has been moved in 6551091d8302e859e842d195c44985e871ed8b6b.

Reply:
I also decided to reinstate the UPPP object that's populated in the context variable as a side-effect of the `page_permissions` tag in a92ed55c3ff8e2adf62c63ac29c34c7012180a5f.

Release notes updated in 2ec1b88e834b616e493e89e0694feb81db16f19e.